config/commands.js

```diff
@@ -676,18 +676,21 @@ var commands = exports.commands = {
 					break;
 
 				case 'moves':
+					var move = {};
+					for (var i in searches[search]) {
+						move[i] = Tools.getMove(i);
+						if (!move[i].exists) return this.sendReplyBox("'" + move[i] + "' is not a known move.");
+					}
 					for (var mon in dex) {
 						var template = Tools.getTemplate(dex[mon].id);
 						if (!template.learnset) template = Tools.getTemplate(template.baseSpecies);
 						if (!template.learnset) continue;
 						for (var i in searches[search]) {
-							var move = Tools.getMove(i);
```

Comment:
it was the function call `getMove` that was slowing it down.
Using a direct reference, instead of dereferencing the property each time makes it faster actually.
And yes, if only `move.id` is used, then you could just say `moveId = moves[i].id`